x-pack/plugins/security/common/licensing/license_service.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Observable, Subscription, of } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ILicense } from '../../../licensing/common/types';
+import { SecurityLicenseFeatures } from './license_features';
+
+export interface SecurityLicense {
+  isEnabled(): boolean;
+  getFeatures(): SecurityLicenseFeatures;
+  features$: Observable<SecurityLicenseFeatures>;
+}
+
+interface SetupDeps {
+  license$: Observable<ILicense>;
+}
+
+export class SecurityLicenseService {
+  private licenseSubscription?: Subscription;
+
+  public setup({ license$ }: SetupDeps) {
+    let rawLicense: Readonly<ILicense> | undefined;
+
+    this.licenseSubscription = license$.subscribe(nextRawLicense => {
+      rawLicense = nextRawLicense;
+    });
+
+    return {
+      license: Object.freeze({
+        isEnabled: () => this.isSecurityEnabledFromRawLicense(rawLicense),
+
+        getFeatures: () => this.calculateFeaturesFromRawLicense(rawLicense),
+
+        features$: license$.pipe(
+          map(nextRawLicense => this.calculateFeaturesFromRawLicense(nextRawLicense))
+        ),
+      }),
+    };
+  }
+
+  public stop() {
+    if (this.licenseSubscription) {
+      this.licenseSubscription.unsubscribe();
+      this.licenseSubscription = undefined;
+    }
+  }
+
+  private isSecurityEnabledFromRawLicense(rawLicense: Readonly<ILicense> | undefined) {
```

Comment:
__note:__ was trying to use new and shiny optional chaining here, but couldn't come up with anything elegant :cry: The closest I have is:
```ts
const securityFeature = rawLicense?.getFeature('security');
return !!(securityFeature?.isAvailable && securityFeature.isEnabled);
```